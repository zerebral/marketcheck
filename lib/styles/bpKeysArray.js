import breakpoints from './breakpoints'

// ['desktop', 'laptop', ...]
export default Object.keys(breakpoints)
