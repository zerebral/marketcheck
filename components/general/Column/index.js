import styled from 'styled-components'
import { rem, clearFix } from 'polished'
import { mediaMin } from '%/styles/mixins'
import { grid, breakpoints } from '%/styles'

const { gutter } = grid

const padding = rem(gutter / 2)

// ['desktop', 'laptop'...]
const bpKeysArray = Object.keys(breakpoints)

const responsiveStyling = props => {
  const usableBreaks = bpKeysArray.filter(bp => props[bp] !== undefined)
  return usableBreaks.map(bp => mediaMin[bp]`
    width: ${props[bp]}%;
  `)
}

const Column = styled.div`
  padding-left: ${padding};
  padding-right: ${padding};
  float: left;
  width: 100%;
  ${clearFix()}

  ${responsiveStyling}
`

export default Column
