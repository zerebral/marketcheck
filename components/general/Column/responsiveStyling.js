import { mediaMin } from '%/styles/mixins'
import { grid, bpKeysArray } from '%/styles'

const percentageCreator = (num, cssKey) => {
  return num
    ? `${cssKey}: ${(num / grid.columns) * 100}%;`
    : undefined
}

// Magic!
const responsiveStyling = props => {
  const usableBreaks = bpKeysArray.filter(bp => props[bp]) // ['laptop']
  return usableBreaks.map(bp => mediaMin[bp]`
    ${percentageCreator(props[bp]['cols'], 'width')}
    ${percentageCreator(props[bp]['offset'], 'margin-left')}
  `)
}

export default responsiveStyling
