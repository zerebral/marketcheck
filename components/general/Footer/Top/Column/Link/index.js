import styled from 'styled-components'
import { rem, em } from 'polished'
import { colors } from '%/styles'

const Link = styled.a`
  display: block;
  color: ${colors.white};
  font-size: ${rem(12)};
  line-height: ${14.4 / 12};
  margin-bottom: ${em(20, 12)};
`

export default Link
