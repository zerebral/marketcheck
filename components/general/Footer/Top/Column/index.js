import styled from 'styled-components'
import { em } from 'polished'
import { grid } from '%/styles'
import { mediaMin } from '%/styles/mixins'

const padding = em(grid.gutter / 2)

const phabletStyles = mediaMin.phablet`
  max-width: 25%;
  flex-basis: 25%;
`

const laptopStyles = mediaMin.laptop`
  display: block;
  max-width: 20%;
  flex-basis: 20%;
  margin-bottom: 0;
`

const Column = styled.div`
  position: relative;
  padding-left: ${padding};
  padding-right: ${padding};
  margin-bottom: ${em(20)};
  max-width: ${props => props.social ? '100%' : '50%'};
  flex-basis: ${props => props.social ? '100%' : '50%'};
  ${phabletStyles}
  ${laptopStyles}
`
export default Column
