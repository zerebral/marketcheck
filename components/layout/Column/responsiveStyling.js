import { mediaMin } from '%/styles/mixins'
import { grid, bpKeysArray } from '%/styles'

/**
 * Take number and cssKey receive percentage
 * @param {Number} num
 * @param {string} cssKey
 *
 * @returns {string}
 */
const percentageCreator = (num, cssKey) => {
  return num
    ? `${cssKey}: ${(num / grid.columns) * 100}%;`
    : undefined
}

/**
 * Takes props from the breakpoints Object and turns them
 * into columns and offsets of that breakpoint
 *
 * @param {Object} props
 * @returns {string}
 */
const responsiveStyling = props => {
  const usableBreaks = bpKeysArray.filter(bp => props[bp]) // ['laptop']
  return usableBreaks.map(bp => mediaMin[bp]`
    ${percentageCreator(props[bp]['cols'], 'width')}
    ${percentageCreator(props[bp]['offset'], 'margin-left')}
  `)
}

export default responsiveStyling
