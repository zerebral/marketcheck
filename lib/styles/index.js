import colors from './colors'
import fonts from './fonts'
import breakpoints from './breakpoints'
import grid from './grid'
import bpKeysArray from './bpKeysArray'

export {
  colors,
  fonts,
  breakpoints,
  bpKeysArray,
  grid
}
