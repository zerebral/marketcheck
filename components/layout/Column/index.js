import styled from 'styled-components'
// Helpers for styling
import { rem, clearFix } from 'polished'
import { grid, bpKeysArray } from '%/styles'
// Helper for the magic
import responsiveStyling from './responsiveStyling'
import PropTypes from 'prop-types'

// Styling helpers
const padding = rem(grid.gutter / 2)

// The component _per se_
const Column = styled.div`
  position: relative;
  padding-left: ${padding};
  padding-right: ${padding};
  float: left;
  width: ${props => props.cols ? `${props.cols / grid.columns * 100}%` : '100%'};
  ${clearFix()}
  ${responsiveStyling}
`

Column.propTypes = {}

// Creative proptyping?
bpKeysArray.map(bp => {
  Column.propTypes[bp] = PropTypes.shape({
    cols: PropTypes.number,
    offset: PropTypes.number
  })
})

export default Column
