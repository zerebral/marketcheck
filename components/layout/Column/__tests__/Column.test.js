/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'
import { breakpoints, grid } from '%/styles'
import Column from '../index'

import 'jest-styled-components'

const percentageCreator = num => `${(num / grid.columns) * 100}%`

describe('Column Component', () => {
  test('Column Passes the snapshot', () => {
    const component = renderer.create(<Column />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Responsive column widths and offsets', () => {
    const component = renderer.create(
      <Column
        phablet={{ offset: 3 }}
        laptop={{ cols: 11, offset: 2 }}
        desktop={{ cols: 10 }} />
    )
    const tree = component.toJSON()

    const expectactions = [
      { rule: 'width', percentage: 10, media: 'desktop' },
      { rule: 'width', percentage: 11, media: 'laptop' },
      { rule: 'margin-left', percentage: 3, media: 'phablet' },
      { rule: 'margin-left', percentage: 2, media: 'laptop' }
    ]

    expectactions.map(e => {
      expect(tree).toHaveStyleRule(e.rule, percentageCreator(e.percentage), { media: `(min-width:${breakpoints[e.media]}px)` })
    })
  })
})
