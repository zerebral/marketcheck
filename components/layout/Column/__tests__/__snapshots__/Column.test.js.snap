// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column Component Column Passes the snapshot 1`] = `
.c0 {
  position: relative;
  padding-left: 0.625rem;
  padding-right: 0.625rem;
  float: left;
  width: 100%;
}

.c0::after {
  clear: both;
  content: "";
  display: table;
}

<div
  className="c0"
/>
`;
